RevenueCatUI/Scripts/PaywallsBehaviour.cs

```diff
@@ -1,32 +1,180 @@
-using System.Threading.Tasks;
+using System;
 using UnityEngine;
+using UnityEngine.Events;
 
 namespace RevenueCatUI
 {
     /// <summary>
-    /// MonoBehaviour helper that forwards to the static PaywallsPresenter API so paywalls can be driven from scenes.
+    /// MonoBehaviour component for presenting RevenueCat paywalls from the Unity Editor.
+    /// Provides an alternative to PaywallsPresenter for developers who prefer configuring
+    /// paywalls through Unity's Inspector interface.
     /// </summary>
+    [AddComponentMenu("RevenueCat/Paywalls Behaviour")]
     public class PaywallsBehaviour : MonoBehaviour
     {
+        [Header("Paywall Options")]
+        [Tooltip("The identifier of the offering to present. Leave empty to use the current offering.")]
+        [SerializeField] private string offeringIdentifier;
+        
+        [Tooltip("Whether to display a close button on the paywall (only for original template paywalls).")]
+        [SerializeField] private bool displayCloseButton = false;
+
+        [Header("Conditional Presentation")]
+        [Tooltip("If set, the paywall will only be presented if the user doesn't have this entitlement.")]
+        [SerializeField] private string requiredEntitlementIdentifier;
+
+        [Header("Events")]
+        [Tooltip("Invoked when the user completes a purchase.")]
```

Comment:
Would it make sense to mention this will only be called once the paywall is dismissed? Like
```suggestion
        [Tooltip("Invoked when the user completes a purchase and the paywall is dismissed.")]
```

Same for the other events. I think it could be confusing otherwise.